### ISSUE TITLE
TransitioningContentControl: old content is not released

### ISSUE THREAD

**Issue by HispidaBytes** [MahApps/MahApps.Metro#3620]
**Describe the bug**
When using a TransitioningContentControl (for example on the HamburgerMenu) the "old" content is not released when the "Transition" property of the TransitioningContentControl is set to "Normal".
It seems that the "AbortTransition" method is never called.

This causes a failure to release the resources used by the view that was previously displayed in the ContentControl.

**Comment by punker76** [comment]
@HispidaBytes Which Version of MahApps? Short sample app?

**Comment by HispidaBytes** [comment]
Ops sorry, I forgot to add some details...
You can use the same repo I've used for an older issue:
[https://github.com/HispidaBytes/HamburgerMenuIssue](https://github.com/HispidaBytes/HamburgerMenuIssue)

I'm using the latest commit 7b61fe8 of MahApps.Metro

What i did was the following:
- Download the latest commit of the MahApps.Metro repo.
- Download the repo linked above.
- Open the "HamburgerMenuTest" solution in Visual Studio and add the "MahApps.Metro" project to the solution as an existing project.
- Reference the added project in the "HamburgerMenuTest" project instead of using the NuGet package.
- Run the "HamburgerMenuTest" project in debug configuration.
- When MainWindow appears, set a breakpoint in the "AbortTransition" method of the class "TransitioningContentControl" of the "MahApps.Metro" project.
- In the MainWindow, move between pages of the HamburgerMenu.
The "AbortTransition" method is never called, so resources of the "previous" view are never released.

**Comment by HispidaBytes** [comment]
Hi, I've been trying to do some debug on this and I may have found something.
In the TransitioningContentControl.xaml file, I've made a duplicate of the "Normal" VisualState and called it "NormalTransition".
Then in the TransitioningContentControl.cs file I've updated the return value of the method "GetTransitionName" to use the new name "NormalTransition".

This seems to fix the problem (the OnTransitionCompleted event handler get called and so the previous view resources are released), but I actually don't understand why completely.
